scripts/jetbrains-remove-node-modules.js

```diff
@@ -2,38 +2,59 @@ const fs = require('fs');
 const path = require('path');
 const os = require('os');
 
-const getAllChildDirectories = (dir) => fs.readdirSync(dir).map(name => path.join(dir, name.toString())).filter(name => fs.lstatSync(name).isDirectory());
+function isDirectory(path) {
+    return fs.lstatSync(path).isDirectory();
+}
+
+const getAllChildDirectories = (dir) => fs.readdirSync(dir).map(name => path.join(dir, name.toString())).filter(isDirectory);
 const isNodeModulesDirectory = (name) => name.toString().endsWith('node_modules');
 
-function getAllNodeModulesPaths(dir) {
+function getAllDirsWithNodeModules(dir) {
     let nodeModulesPaths = [];
     getAllChildDirectories(dir).forEach(name => {
         if (isNodeModulesDirectory(name)) {
             console.log('Excluding ' + name);
-            nodeModulesPaths.push(name);
+            nodeModulesPaths.push(dir);
         } else {
-            const subNodeModulesPaths = getAllNodeModulesPaths(name);
+            const subNodeModulesPaths = getAllDirsWithNodeModules(name);
             nodeModulesPaths = nodeModulesPaths.concat(subNodeModulesPaths);
         }
     });
     return nodeModulesPaths;
 }
 
+function isCdkPackageDirectory(path) {
+    return fs.existsSync(path + '/lib') && isDirectory(path + '/lib') &&
+        fs.existsSync(path + '/package.json');
+}
+
+function cdkPackageEntries(path) {
+    let ret = [`<excludeFolder url="file://$MODULE_DIR$/${path}/node_modules" />`];
+    if (isCdkPackageDirectory(path)) {
+        ret.push(`<excludeFolder url="file://$MODULE_DIR$/${path}/.nyc_output" />`);
+        ret.push(`<excludeFolder url="file://$MODULE_DIR$/${path}/coverage" />`);
+        ret.push(`<sourceFolder url="file://$MODULE_DIR$/${path}/lib" isTestSource="false" />`);
+        ret.push(`<sourceFolder url="file://$MODULE_DIR$/${path}/test" isTestSource="true" />`);
+    }
+    return ret.join(os.EOL);
+}
+
 // Should be run at the root directory
 if (!fs.existsSync('lerna.json')) {
     throw new Error('This script should be run from the root of the repo.');
 }
 
-const nodeModulesPaths = getAllNodeModulesPaths('.');
+const nodeModulesPaths = getAllDirsWithNodeModules('.');
 
 // Hardcoded exclusions for this project (in addition to node_modules)
-const exclusions = nodeModulesPaths.map(path => `<excludeFolder url="file://$MODULE_DIR$/${path}" />`);
-exclusions.push('<excludeFolder url="file://$MODULE_DIR$/.tmp" />');
+const exclusions = nodeModulesPaths.map(cdkPackageEntries);
+exclusions.push('<excludeFolder url="file://$MODULE_DIR$/.github" />');
+exclusions.push('<excludeFolder url="file://$MODULE_DIR$/.idea" />');
+exclusions.push('<excludeFolder url="file://$MODULE_DIR$/.tools" />');
 exclusions.push('<excludeFolder url="file://$MODULE_DIR$/docs" />');
-exclusions.push('<excludeFolder url="file://$MODULE_DIR$/temp" />');
```

Comment:
The regex replaces the content of content wholesale. temp/tmp/.tmp are the defaults when creating the project in intellij. I think we should probably keep those in there.

Reply:
But we don't use those directories in the CDK, right? So those entries will never actually ignore anything. I always remove them manually after regenerating the file, so that's why I actually removed them.